File: frontend/demo/component/passwordfield/password-field-styles.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/password-field';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('password-field-styles')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-password-field
        theme="align-right small helper-above-field"
        label="Label"
        helper-text="Helper text"
        value="Ex@mplePassw0rd"
        style="--vaadin-input-field-border-width: 1px;"
      ></vaadin-password-field>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/passwordfield/PasswordFieldStyles.java
package com.vaadin.demo.component.passwordfield;

import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.textfield.PasswordField;
import com.vaadin.flow.component.textfield.TextFieldVariant;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("password-field-styles")
public class PasswordFieldStyles extends HorizontalLayout {

    public PasswordFieldStyles() {
        setPadding(false);

        // tag::snippet[]
        PasswordField field = new PasswordField();
        field.addThemeVariants(
            TextFieldVariant.LUMO_SMALL,
            TextFieldVariant.LUMO_ALIGN_RIGHT,
            TextFieldVariant.LUMO_HELPER_ABOVE_FIELD
        );
        field.getStyle().set("--vaadin-input-field-border-width", "1px");
        // end::snippet[]
        field.setLabel("Label");
        field.setHelperText("Helper text");
        field.setValue("Ex@mplePassw0rd");

        add(field);
    }

    public static class Exporter extends DemoExporter<PasswordFieldStyles> { // hidden-source-line
    } // hidden-source-line
}
